### PR TITLE
Adding changes to provide descriptions of schemas

### PR DESCRIPTION
**Pull request by antoniogarrote**
Minor changes:
- Updating AMF to 3.5.0
- Passing an input directory as an argument

Now we are collecting also vocabulary files and building an ontology index with the URIs and descriptions for each term.
When parsing dialects, the classTerm and propertyTerms mapped in the shapes and property mappings are looked up in the ontology index and the description applied to the schema.

### CODE REVIEW

File: src/data_collectors.js [hunk 1]
```diff
@@ -1,7 +1,32 @@
 const utils = require('./utils')
 
+/* Collects complete vocabulary data. */
+function collectVocabularyData (doc, ctx, acc) {
+  const id = doc.query('@id')
+  const dialectData = {
```
Comment (postatum) [inline src/data_collectors.js]: Is this a correct var name? Is this really a dialect data?

File: src/data_collectors.js [hunk 2]
```diff
@@ -17,14 +43,90 @@ function collectDialectData (doc, ctx, acc) {
...
+    dialectData.nodeMappings = collectNodesData(
+      doc, dialectData, ctx, ontologyTerms)
       .sort(utils.nameSorter)
     return dialectData
   }
 }
 
 /* Collects dialect nodeMappings data. */
-function collectNodesData (doc, dialectData, ctx) {
+function collectVocabularyNodesData (doc, dialectData, ctx) {
```
Comment (postatum) [inline src/data_collectors.js:54]: Please update a docstring and comment above each `doc.queryAll` what that block collects data for.

File: src/data_collectors.js [hunk 3]
```diff
@@ -80,22 +188,28 @@ function collectNodesData (doc, dialectData, ctx) {
...
   return propsNodes.map(prop => {
-    const propData = collectCommonPropData(doc, prop)
+    const propData = collectCommonPropData(doc, prop, ontologyTerms)
     propData.range = utils.parseHashValue(prop.query('shacl:datatype @id'))
     return propData
   })
 }
 
+/* Collects super-classes for a class */
+function collectSubClassOfLinks (doc, node) {
```
Comment (postatum) [inline src/data_collectors.js]: Is this function implemented? It's description says it should "collect super-classes for a class" but query it performs doesn't use any arguments. Also `doc` is not used.

File: src/data_collectors.js [hunk 4]
```diff
@@ -177,6 +295,22 @@ function collectCommonNavData (dialectsData) {
   return commonNavData
 }
 
+function sortProps (props) {
```
Comment (postatum) [inline src/data_collectors.js]: This belongs to `utils.js`. Also sorting function from line 304 is already implemented in `utils.nameSorter`.

File: src/index.js [hunk 5]
```diff
@@ -33,27 +74,46 @@ async function main () {
...
+  const ontologyTerms = {}
+  Object.values(ontology).forEach(function (vocab) {
+    vocab.nodeMappings.forEach(function (term) {
+      ontologyTerms[term.id] = term
+    })
+  })
+
+  // Let's process the dialects
+  for (var o in jsonGraph) {
+    const graph = jsonGraph[p]
```
Comment (postatum) [inline src/index.js:114]: The loop defines var `o`, but uses var `p` from a previous loop.

File: src/utils.js [hunk 6]
```diff
@@ -135,7 +136,24 @@ function loadConfig (cfgName, ctx) {
   return ctx
 }
 
+function walkSync (dir, filelist) {
+  var files = fs.readdirSync(dir)
+  filelist = filelist || []
+  files.forEach(function (file) {
+    const stats = fs.lstatSync(dir + '/' + file)
```
Comment (postatum) [inline src/utils.js]: You can use `path.join(dir, file)` instead of `dir + '/' + file`.

File: templates/index.mustache [hunk 7]
```diff
@@ -14,77 +14,77 @@
...
+	</div>
       </div>
 
 
       <div class="col-md-10 content-block" id="index-content-block">
 
-        <div class="index-dialects-section">
-          <h1>Dialects</h1>
+	<div class="index-dialects-section">
+	  <h1>Cloud Information Model 1.0-SNAPSHOT</h1>
```
Comment (postatum) [inline templates/index.mustache]: What's this `Cloud Information Model 1.0-SNAPSHOT` ?

**Review by postatum — CHANGES_REQUESTED** [review]
See comments.
Please run `standard --fix src/*` after you've done working on the code.

I've also pushed some style fixes to the branch.

File: src/data_collectors.js (hunk 2 above)
Comment (antoniogarrote) [reply src/data_collectors.js:54]: Added!

**Comment by jstoiko** [comment]
> Please run `standard --fix src/*` after you've done working on the code.

@postatum: `standard` is not a dev dependency for some reason. maybe it should be there and [setup in one of the `"scripts"`](https://www.npmjs.com/package/standard#what-you-might-do-if-youre-clever).

**Comment by antoniogarrote** [comment]
@postatum Please, review changes.
Thanks for the comments

**Comment by postatum** [comment]
> > Please run `standard --fix src/*` after you've done working on the code.
> 
> @postatum: `standard` is not a dev dependency for some reason. maybe it should be there and [setup in one of the `"scripts"`](https://www.npmjs.com/package/standard#what-you-might-do-if-youre-clever).

I've added it as a dev dependency in this PR branch. 
I thought about adding it to pre-commit hook, but didn't figure out how yet. Not sure if I need more rights for that.

**Review by postatum — APPROVED** [review]
Looks good now.

**Pull request merged**